services/src/main/java/org/fao/geonet/api/records/MetadataWorkflowApi.java

```diff
@@ -553,6 +561,126 @@ public MetadataStatus convertParameter(int id, MetadataStatusParameter parameter
+        }
+
+        String metadataId = metadataStatusId.getMetadataId() + "";
+
+        try {
+            ApiUtils.canViewRecord(metadataId, request);
+        } catch (SecurityException e) {
+            Log.debug(API.LOG_MODULE_NAME, e.getMessage(), e);
+            throw new NotAllowedException(ApiParams.API_RESPONSE_NOT_ALLOWED_CAN_VIEW);
+        }
+
+        return status.getPreviousState();
+    }
+
+
+    @ApiOperation(
+            value = "Restore saved content from a status record",
+            notes = "",
+            nickname = "restoreAtStatusSave")
+    @RequestMapping(
+            value = "/status/{serializedId}/restore",
+            method = RequestMethod.POST
+            )
+    @PreAuthorize("hasRole('Editor')")
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "Record restored."),
+            @ApiResponse(code = 403, message = ApiParams.API_RESPONSE_NOT_ALLOWED_CAN_EDIT)
+    })
+    @ResponseStatus(HttpStatus.OK)
+    @ResponseBody
+    public void restoreAtStatusSave(
+            @ApiParam(
+                    value = "Record status id",
+                    required = true)
+            @PathVariable
+            String serializedId,
+            HttpServletRequest request
+            )
+                    throws Exception {
+        ApplicationContext applicationContext = ApplicationContextHolder.get();
+        MetadataStatusId metadataStatusId = MetadataStatusResponse.convertSerializedStringInMetadataStatusId(serializedId);
+        IMetadataManager dataMan = applicationContext.getBean(IMetadataManager.class);
+
+        MetadataStatusRepository metadataStatusRepository =
+                ApplicationContextHolder.get().getBean(MetadataStatusRepository.class);
+
+        MetadataStatus status = metadataStatusRepository.findOne(metadataStatusId);
+
+        if(!StatusValueType.event.equals(status.getStatusValue().getType()) || !ArrayUtils.contains(supportedStatuses, metadataStatusId.getStatusId())) {
+            throw new NotAllowedException("The recover for this element is not supported");
+        }
+
+        String metadataId = metadataStatusId.getMetadataId() + "";
+
+        AbstractMetadata metadata = null;
+        try {
+            metadata = ApiUtils.canEditRecord(metadataId, request);
+        } catch (SecurityException e) {
+            Log.debug(API.LOG_MODULE_NAME, e.getMessage(), e);
+            throw new NotAllowedException(ApiParams.API_RESPONSE_NOT_ALLOWED_CAN_EDIT);
+        }
+        Locale locale = languageUtils.parseAcceptLanguage(request.getLocales());
+        ServiceContext context = ApiUtils.createServiceContext(request, locale.getISO3Language());
+
+        String xmlBefore = metadata.getData();
+        Element md = Xml.loadString(status.getPreviousState(), false);
+        dataMan.updateMetadata(context, metadataId, md, false, true, true, context.getLanguage(), null, true);
```

Comment:
Probably requires to check also about the validation info in the validation table if exists for the metadata that is being restored.

Reply:
I have the impression that any operation that changes the content of a metadata record ignores the presence of a validation record associated (at least basic editor operations). I don't know if it has to be different.